### PR TITLE
Add "snapshot-count" annotation for volumesnapshotClass

### PR DESCRIPTION
**Pull request by f10atin9**
Signed-off-by: f10atin9 <f10atin9@kubesphere.io>

### What type of PR is this?

/kind feature

### What this PR does / why we need it:
In the console, the snapshotclass list is required to display the number of snapshots.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
None
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go [hunk 1]
```diff
@@ -53,6 +56,11 @@ func (v *volumeSnapshotClassGetter) List(namespace string, query *query.Query) (
 
 	var result []runtime.Object
 	for _, snapshotClass := range all {
+		count := v.countVolumeSnapshots(snapshotClass.Name)
+		if snapshotClass.Annotations == nil {
+			snapshotClass.Annotations = make(map[string]string)
```
Comment (dkeven) [inline pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go:61]: Do a deepcopy before mutating the SnapshotClass, as in:
https://github.com/kubesphere/kubesphere/pull/4599

**Review by dkeven — CHANGES_REQUESTED** [review]

File: pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go (hunk 1 above)
Comment (f10atin9) [reply pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go:62]: Okay, I've added it.

File: pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go [hunk 2]
```diff
@@ -86,3 +95,18 @@ func (v *volumeSnapshotClassGetter) filter(object runtime.Object, filter query.F
 		return v1alpha3.DefaultObjectMetaFilter(snapshotClass.ObjectMeta, filter)
 	}
 }
+
+func (v *volumeSnapshotClassGetter) countVolumeSnapshots(name string) int {
+	snapshots, err := v.informers.Snapshot().V1().VolumeSnapshots().Lister().VolumeSnapshots("").List(labels.Everything())
```
Comment (stoneshi-yunify) [inline pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go:100]: is this correct? shouldn't be `v.informers.Snapshot().V1().VolumeSnapshots().Lister().List(labels.Everything())`?

Comment (f10atin9) [reply pkg/models/resources/v1alpha3/volumesnapshotclass/volumesnapshotclass.go]: My mistake, it has been changed.

**Comment by dkeven** [comment]
The unit tests should be changed accordingly, too. As it used objects from the altered cache to do assertions before, which is wrong.

**Pull request merged**